EIPS/eip-4519.md

```diff
@@ -0,0 +1,239 @@
+
+## Specification
+This kind of non-fungible token is designed to represent a secure smart asset with the capacity of generating its own account and, hence, interacting with Ethereum chain. Besides, this SmartNFT can assign the token to a user and allows the establishment of secure communication channels with the owner and user. The following attributes are proposed for the token:
+
+| Type | Name of variable | Defined by ERC-721 | Optional |
+|--|--|--|--|
+| uint256 | tokenId | Yes | No |
+| address | owner | Yes | No |
+| address | approved | Yes | Yes |
+| address | approvedForAll | Yes | Yes |
+| address | asset | No | Yes |
+| address | user | No | Yes |
+| enum | state | No | No |
+| uint256 | hashK_OD | No | Yes |
+| uint256 | HashK_UD | No | Yes |
+| uint256 | dataEngagement | No | Yes |
+| uint256 | timeStamp | No | No |
+| uint256 | timeout | No | No |
+
+The atTributes do not need to be given in a specific order. The designer of the smart contract defines the order. Like in the ERC-721, the role of operator (approvedForAll) is an attribute more related to the owner than to the token.
+ 
+Asset and user are optional attributes but at least one of them should be used in a SmartNFT. In the case of using only the attribute user, two states define if the token is assigned or not to a user. Figure 1 shows the corresponding states in a flow chart. When a token is created, transferred or unassigned, the token state is set to “notAssigned”. If the token is assigned to a valid user, the state is set to "userAssigned".
+ 
+![Figure 1 : Flow chart of the token states with user defined (and asset undefined)](../assets/eip-4519/images/Figure1.png)
+  
+In the case of defining the asset attribute but not the user attribute, two states define if the token is waiting for authentication with the owner or authentication has finished successfully. Figure 2 shows the corresponding states in a flow chart. When a token is created or transferred to a new owner, then the token changes its state to "waitingForOwner". In this state, the token is waiting for authentication by the owner. Once the asset is authenticated, its associated token changes its state to "engagedWithOwner".
```

Comment:
```suggestion
In the case of defining the asset attribute but not the user attribute, two states define if the token is waiting for authentication with the owner or **if** authentication has finished successfully. Figure 2 shows the corresponding states in a flow chart. When a token is created or transferred to a new owner, then the token changes its state to "waitingForOwner". In this state, the token is waiting for authentication by the owner. Once the asset is authenticated, its associated token changes its state to "engagedWithOwner".
```

Reply:
@MicahZoltu potential spammer, suggsting irrelavant changes!

Reply:
Thanks, please report to GitHub as well using the `...` menu.